portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,31 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Cycles through images automatically.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function slideShow(imgIndex = 0) {
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  const slides = document.getElementsByClassName('my-slides');
+  
+  for (let i = 0; i < slides.length; i++) {
+    slides[i].style.display = 'none';
+  }
+  
+  // If there are no images, end function.
```

Comment:
lets move this check above the loop on line 22 - you won't ever enter that loop either if length == 0 so its cleaner to short circuit out of the method as quickly as possible